catalog_integrations_api_tools/parser.py

```diff
@@ -0,0 +1,297 @@
+import json
+import logging
+import os
+import re
+from dataclasses import dataclass
+from typing import Optional
+
+import pandas as pd
+from numpy import nan
+
+
+@dataclass
+class AvailabilityStatus:
+    available: int = 1
+    unavailable: int = 0
+    out_of_stock: int = 0
+
+    def __init__(self, available: int = 1, unavailable: int = 0, out_of_stock: int = 0):
+        if (
+            not isinstance(available, int)
+            or not isinstance(unavailable, int)
+            or not isinstance(out_of_stock, int)
+        ):
+
+            raise TypeError('Arguments should all be of type "int"')
+        self.available = available
+        self.unavailable = unavailable
+        self.out_of_stock = out_of_stock
+
+
+class Parser:
+    currencies = {
+        "CAD": 100,
+        "USD": 100,
+        "CLP": 100,
+        "MXN": 100,
+        "BRL": 100,
+        "COP": 100,
+        "PEN": 100,
+    }
+    logger = logging.getLogger(__name__)
+    base_units = ["LB", "KG", "UN"]
+    tags = ["PRESCRIPTION", "ALCOHOL"]
+    products_filename = "products.csv"
+    stock_prices_filename = "stock-prices.csv"
+
+    def __init__(
+        self,
+        read_directory: str,
+        write_directory: str,
+        file_prefix: str,
+        currency: Optional[str] = None,
+        products: bool = True,
+        stock_prices: bool = True,
+        status: AvailabilityStatus = AvailabilityStatus(),
+    ):
+        self._read_directory = read_directory
```

Comment:
I would prefer to have this value as optional and handled the logic to the default value is not defined
```suggestion
        status: Optional[AvailabilityStatus] = None,
    ):
        if status is None:
            status = AvailabilityStatus()
        self._read_directory = read_directory
```